### PR TITLE
Add CA certificate configuration details to template

### PR DESCRIPTION
**Pull request by JD-Robertson**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/install-systemlink-enterprise/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Update the values template with more details related to the configuration of custom CA certificates.

### Why should this Pull Request be merged?

Necessary documentation.

### What testing has been done?

N/A

### CODE REVIEW

File: getting-started/templates/systemlink-values.yaml [hunk 1]
```diff
@@ -92,11 +92,39 @@ global:
...
+  ##  - secretName: "ca-secret"
+  ##    key: "cert"
+  ##  - secretName: "ca-config-map"
+  ##    key: "my-cert"
+  ##    isConfigMap: true
+  ##
+  #  <ATTENTION> - If one or more custom Certificate Authorities are required for HTTPS access to this application
+  #                or for TLS access to any external resources referenced by this application (databases, file storage, etc.),
+  #                a copy of the certificate chain must be deployed as a secret to the cluster prior to deploying this application.
+  #                For certificate authorities, only the public certificate needs to be deployed to the cluster.
```
Comment (prestwick) [inline getting-started/templates/systemlink-values.yaml:106]: I don't understand exactly what we are communicating in this line. I could see this meaning one of two things. 

- For deployments using certificates signed by a public certificate authority, administrator need not also deploy the root certificate or certificate chain using this configuration.

OR 

- "For certificate authorities only the root public certificate needs to be deployed to the cluster" e.g. the entire trust chain ins't needed, just the root.

File: getting-started/templates/systemlink-values.yaml [hunk 2]
```diff
@@ -92,11 +92,39 @@ global:
...
+  #
+  #                EXAMPLE:  kubectl -n systemlink-namespace create secret generic my-ca-certificate --from-file=cert=path/to/my-ca.crt
+  #
+  #                Use the 'trustedCertificatesSecrets', 'apiHostCertificateSecret' and 'deployApiHostCertificateToSystems'
+  #                values below to configure the application and registered client systems with the required trust chain.
+  #
   trustedCertificatesSecrets: []
+  ##
+  ## Secret containing the TLS certificate for the API host. This configuration is only necessary when the API
+  ## host is using a custom root CA certificate. This certificate will be deployed and installed on all managed clients
```
Comment (prestwick) [inline getting-started/templates/systemlink-values.yaml:116]:
```suggestion
  ## host is using a private root CA certificate. This certificate will be deployed and installed on all managed clients
```

File: getting-started/templates/systemlink-values.yaml [hunk 3]
```diff
@@ -92,11 +92,39 @@ global:
...
+  ## to enable the communication with the API host. Only valid if deployApiHostCertificateToSystems is true.
```
Comment (prestwick) [inline getting-started/templates/systemlink-values.yaml:117]: I think we can combine these last two sentences to have this read better, assuming I'm not changing what we're trying to communicate here. 

"When deployApiHostCertificateToSystems is true, this certificate will be deployed and installed on all managed clients to enable communication with the API host."

File: getting-started/templates/systemlink-values.yaml [hunk 4]
```diff
@@ -92,11 +92,39 @@ global:
...
+  ##
+  apiHostCertificateSecret:
+  #  secretName: "ca-secret"
+  #  key: "certificate"
+  ##
+  ## Deploy the API host CA root certificate to all managed systems. This configuration is necessary when the API host
```
Comment (prestwick) [inline getting-started/templates/systemlink-values.yaml:123]: I wonder if we can just leave out these last three lines. They are basically describing again what was described and referenced just above. I think leaving this out could make these docs more clear and readable.

**Review by prestwick — CHANGES_REQUESTED** [review]
Some suggests for clarity.

File: getting-started/templates/systemlink-values.yaml (hunk 1 above)
Comment (JD-Robertson) [reply getting-started/templates/systemlink-values.yaml:106]: Neither. I don't want them to deploy the private key data for the certificate. Just the public certificate.

File: getting-started/templates/systemlink-values.yaml [hunk 5]
```diff
@@ -92,11 +92,37 @@ global:
...
+  #
+  #                EXAMPLE:  kubectl -n systemlink-namespace create secret generic my-ca-certificate --from-file=cert=path/to/my-ca.crt
+  #
+  #                Use the 'trustedCertificatesSecrets', 'apiHostCertificateSecret' and 'deployApiHostCertificateToSystems'
+  #                values below to configure the application and registered client systems with the required trust chain.
+  #
   trustedCertificatesSecrets: []
+  ##
+  ## Secret containing the TLS certificate for the API host. This configuration is only necessary when the API
+  ## host is using a custom root CA certificate. When deployApiHostCertificateToSystems is true, this certificate
```
Comment (prestwick) [inline getting-started/templates/systemlink-values.yaml:116]:
```suggestion
  ## host is using a private root CA certificate. When deployApiHostCertificateToSystems is true, this certificate
```

**Review by prestwick — APPROVED** [review]

**Pull request merged**